### PR TITLE
render enemies in enemy mode

### PR DESCRIPTION
**Pull request by tolmar**
This change makes it so the enemy viewer parses enemy data deep enough to show enemy groups.

It uses quite a bit of semantic zoom stuff to make the data actually visible and pleasant on the map, so the rendering code is a little numbery.

### CODE REVIEW

File: src/ebhack/MapDisplay.java [hunk 1]
```diff
@@ -475,7 +475,96 @@ private void drawMap(Graphics2D g) {
...
+        g.setComposite(AlphaComposite.getInstance(
+                AlphaComposite.SRC_OVER, 0.75F));
+        g.setPaint(enemyColors[mapEnemyGroupId]);
+        g.fill(rect);
+        g.setComposite(AlphaComposite.getInstance(
+                AlphaComposite.SRC_OVER, 1.0F));
+
+        // Draw the enemies in a clip region so it's okay to overflow
+        if (zoom >= MIN_ZOOM_TO_DISPLAY_ENEMIES) {
+            g.setClip(rect);
```
Comment (brentjs0) [inline src/ebhack/MapDisplay.java]:
```suggestion
            Rectangle2D originalClip = (Rectangle2D) g.getClip();
            g.setClip(originalClip.createIntersection(rect));
```

File: src/ebhack/MapDisplay.java [hunk 2]
```diff
@@ -475,7 +475,96 @@ private void drawMap(Graphics2D g) {
...
+                int y = (int) (ENEMY_ROW_2_Y * enemySpriteZoom());
+                for (MapEnemyGroup.SpawnGroup spawnGroup : mapEnemyGroup.subGroup2) {
+                    x = 4 + drawEnemyGroup(g, rect, new Point(x, y), spawnGroup.probability, spawnGroup.enemyGroup);
+                }
+                String spawnRate = mapEnemyGroup.subGroup2Rate + "%";
+                Rectangle2D textBG = g.getFontMetrics().getStringBounds(spawnRate, g);
+                g.setPaint(Color.white);
+                g.drawString(spawnRate, (int) (rect.getMaxX() - textBG.getWidth()), (int) rect.getY() + 48);
+            }
+            g.setClip(null);
```
Comment (brentjs0) [inline src/ebhack/MapDisplay.java]:
```suggestion
            g.setClip(originalClip);
```

**Review by brentjs0 — COMMENTED** [review]
These are the changes I made to prevent the enemy groups from being drawn on top of the other UI elements.

**Review by brentjs0 — APPROVED** [review]
Looks good. Awesome work.

**Comment by tolmar** [comment]
Thanks! Woohoo!

**Pull request merged**